Update public docs for preview period (#240)
apollographql/apollo-cli: docs/source/essentials.md

## Graph vs Subgraph

> Rover uses the terms `graph` and `subgraph` to differentiate between a monolithic,
> traditional GraphQL endpoint and a federated graph following Apollo's federation
> spec, respectively. Generally, when working with federated graphs, you will be running
> operations based on a single _subgraph_ (and thus using the `subgraph` command), 
> rather than on the whole composed graph.

Reviewer: ```suggestion
Rover uses the terms `graph` to refer to traditional, monolithic graphs, and `subgraph` to refer to each "subgraph" that is a part of a larger [federated graph](https://www.apollographql.com/docs/federation/). Generally, when working with federated graphs, you will be running
operations based on a single _subgraph_ (and thus using the `subgraph` command), 
rather than on the whole composed graph.
```

maayybbe a bit better though I still dont love it - just trying to work on that first sentence since there's a bunch to think about.

Author: Definitely better! I added that link. Feel free to iterate more if you see any better way to word it!

### Graph Refs

> Rover uses a format known as a graph ref to refer to a graph id/variant combination.
> This format defines a graph ref as `graph_id@variant_name`. All rover commands that
> interact with the Apollo graph registry require a graph ref as their first positional
> argument. If using the default variant, `current`, you don't need to provide the
> variant in the graph ref, although it is recommended for clarity.

Reviewer: ```suggestion
Rover uses a format known as a graph ref to refer to a graph id/variant combination.
This format defines a graph ref as `graph_id@variant_name`. All Rover commands that
interact with the Apollo graph registry require a graph ref as their first positional
argument. If using the default variant, `current`, you don't need to provide the
variant in the graph ref, although it is recommended for clarity.
```
